### PR TITLE
Add Support of changing user status

### PR DESCRIPTION
**Pull request by robbix1206**
Recently Discord add the support to change the User status beetween Online, Idle, Do not Distrub and Invisible.
This PR is to implement that new functionality.

### CODE REVIEW

File: restapi.go [hunk 1]
```diff
@@ -320,6 +320,23 @@ func (s *Session) UserSettings() (st *Settings, err error) {
 	return
 }
 
+// UserUpdateStatus update the user status
+// status   : The new status (Actual valid status are 'online','idle','dnd','invisible')
+func (s *Session) UserUpdateStatus(status string) (st *Settings, err error) {
```
Comment (iopred) [inline restapi.go]: Consider making a type and constants for the valid values, like VerificationLevel, so that we can restrict the values of status.

Comment (robbix1206) [inline restapi.go]: The problem is actually you have 5 status.
"online","idle","dnd","invisible","offline".
However the Status can be set by the user as "online","idle","dnd" or "invisible"
And the library can see only Users as "online","idle","dnd" or "offline"
So even if there is a Status type the user can send a status bad type ("offline")
I will do that and make a verification in the function.

**Comment by robbix1206** [comment]
So Status is now a type.
Look like it works ^^.

File: structs.go [hunk 2]
```diff
@@ -254,9 +254,9 @@ type VoiceState struct {
 
 // A Presence stores the online, offline, or idle and game status of Guild members.
 type Presence struct {
-	User   *User  `json:"user"`
-	Status string `json:"status"`
-	Game   *Game  `json:"game"`
+	User   *User   `json:"user"`
+	Status *Status `json:"status"`
```
Comment (iopred) [inline structs.go]: Status Status `json:"status"`

File: structs.go [hunk 3]
```diff
@@ -292,20 +292,34 @@ type User struct {
...
+	EnableTtsCommand       bool               `json:"enable_tts_command"`
+	MessageDisplayCompact  bool               `json:"message_display_compact"`
+	ShowCurrentGame        bool               `json:"show_current_game"`
+	ConvertEmoticons       bool               `json:"convert_emoticons"`
+	Locale                 string             `json:"locale"`
+	Theme                  string             `json:"theme"`
+	GuildPositions         []string           `json:"guild_positions"`
+	RestrictedGuilds       []string           `json:"restricted_guilds"`
+	FriendSourceFlags      *FriendSourceFlags `json:"friend_source_flags"`
+	Status                 *Status            `json:"status"`
```
Comment (iopred) [inline structs.go]: Status instead of *Status

Comment (robbix1206) [inline structs.go]: Ok will fix it

**Pull request merged**